index.ts

```diff
@@ -1,35 +1,60 @@
 export class OatyArray {
-  private _original: object[]
-  private _keys: string[]
-  private _transposed: object = {}
+  private _keys: string[] | undefined
+  private _original: object[] = []
+  private _transposed: { [key: string]: any } = {}
+
+  constructor(data?: object[], options?: { keys?: string[] }) {

```

Comment:
I think I may revert back to having `data` inside `options`.
I don't really want to have to specify initialising with an empty array, and I can't do 
`OatyArray({keys: [...]})` because it thinks I'm trying to set data to `{keys:[...]}` which doesn't work

Reply:
Reverted